### ISSUE TITLE
Add editorconfig support for `is null` instead of == null (and `not` variant)

### ISSUE THREAD

**Issue by NTaylorMullen** [dotnet/roslyn#65572]
**Brief description:**
Using `is null` instead of `== null` can be preferable in order to avoid operator overloads for `==`. Many codebases are now adopting this pattern.

Describe your **code style** rule here: Wherever you see `==` null prefer `is null`, wherever you see `!= null` prefer `is not null`.

**Languages applicable:** C#

**Code example that the analyzer should report:**

```C#
Func<bool>? method = TryGetSomeMethod();

if (method != null && !method())
{
    Console.WriteLine("Returned false");
}
```

**Comment by Youssef1313** [comment]
This would be a generalization of IDE0041, I think

**Comment by Youssef1313** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/56859

https://github.com/dotnet/roslyn/issues/56859#issuecomment-940455225

**Comment by sharwell** [comment]
Duplicate of #56859
